app/gui/src/model/project/synchronized.rs

```diff
@@ -167,6 +169,44 @@ impl ContentRoots {
 }
 
 
+
+// ========================
+// === VCS status check ===
+// ========================
+
+/// Check whether the current state of the project differs from the most recent snapshot in the VCS,
+/// and emit a notification.
+#[profile(Detail)]
+async fn check_vcs_status_and_notify(
+    project_root_id: Uuid,
+    language_server: Rc<language_server::Connection>,
+    publisher: notification::Publisher<model::project::Notification>,
+) -> json_rpc::Result<language_server::response::VcsStatus> {
+    let path_segments: [&str; 0] = [];
+    let root_path = language_server::Path::new(project_root_id, &path_segments);
+    let status = language_server.vcs_status(&root_path).await;
+    match &status {
+        Err(_) => {
+            publisher.notify(model::project::Notification::VcsStatusChanged(
+                model::project::VcsStatus::Dirty,
+            ));
+        }
+        Ok(language_server::response::VcsStatus { dirty: true, .. }) => {
+            publisher.notify(model::project::Notification::VcsStatusChanged(
+                model::project::VcsStatus::Dirty,
+            ));
+        }
+        Ok(language_server::response::VcsStatus { dirty: false, .. }) => {
+            publisher.notify(model::project::Notification::VcsStatusChanged(
+                model::project::VcsStatus::Clean,
+            ));
+        }
+    }
```

Comment:
We can deduplicate things here somehow by extracting notify call and notification constructing:
```rust
let status = match &status {...}
publisher.notify(model::project::Notification::VcsStatusChanged(status);
```

Reply:
Good point, I didn't see it. Fixed now.